src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -73094,6 +73094,124 @@ void M1()
                 );
         }
 
+        [Fact]
+        public void TypeArgumentInference_09()
+        {
+            var source = @"
+class C
+{
+    T M<T>(T t1, T? t2) => t2 ?? t1;
```

Comment:
>T M<T>(T t1, T? t2) => t2 ?? t1; [](http://example.com/codeflow?start=4&length=32)

Consider testing with `T!` and `T~` as well.

Reply:
Adding the test, but not intending to modify the implementation for this scenario unless it feels really bad/unusable. Feels tangentially related to #56518 (what should happen to type arguments when type parameters are oblivious).

I think tests for not-annotated type parameters are already present. For example, `TypeArgumentInference_02`.